src/org/exist/backup/ConsistencyCheck.java

```diff
@@ -322,8 +319,8 @@ public ErrorReport checkXMLTree( final DocumentImpl doc )
                     public Object start() {
                         EmbeddedXMLStreamReader reader = null;
                         try {
-                            final ElementImpl             root            = (ElementImpl)doc.getDocumentElement();
-                            reader = (EmbeddedXMLStreamReader)broker.getXMLStreamReader( root, true );
+                            final Node              root            = doc.getFirstChild();
```

Comment:
First child.... should we do this for all children? Is a comment also a node ?

Reply:
First child of the document is just the starting point for the scan. The XMLStreamReader will then read over all other nodes. So yes, the first node could be a comment.